### ISSUE TITLE
[bug] `no-context-provider` false positive for third party libs

### ISSUE THREAD

**Issue by songkeys** [Rel1cx/eslint-react#1295]
### Describe the bug

```
import { Tooltip } from '@base-ui-components/react/tooltip'


<Tooltip.Provider> // <- false positive
```

### Platform and versions

```Text
`"@eslint-react/eslint-plugin": "^2.2.3",`
```

### Stack trace

```Text

```
